src/main/java/world/bentobox/bentobox/managers/IslandsManager.java

```diff
@@ -1520,8 +1520,10 @@ public void setOwner(World world, User user, UUID targetUUID) {
      */
     public void setOwner(User user, UUID targetUUID, Island island) {
         islandCache.setOwner(island, targetUUID);
-        // Remove the old owner from the island
+        // Remove the old owner from the island to clear cache
         plugin.getIslands().removePlayer(island, user.getUniqueId());
```

Comment:
After looking at the code of `removePlayer`, I don't think this should be done. The user should still remain in the islandCache (in islandsByUUID) even if they are just a team member.